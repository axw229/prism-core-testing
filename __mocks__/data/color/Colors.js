// @flow
import cloneDeep from 'lodash/cloneDeep'
import sample from 'lodash/sample'

import type { ColorSetPayload, Color } from '../../../src/shared/types/Colors'

export const Colors: ColorSetPayload = {
  "Red": [
    [
      {
        "colorNumber": "6561",
        "coordinatingColors": {
          "coord2ColorId": 1926,
          "coord1ColorId": 2250,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2248",
        "isExterior": true,
        "isInterior": true,
        "name": "Teaberry",
        "rgb": 15454683,
        "lrv": 68.551,
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 235,
        "green": 209,
        "blue": 219,
        "hue": 0.935897435897436,
        "saturation": 0.39393939393939387,
        "lightness": 0.8705882352941177,
        "hex": "#ebd1db",
        "isDark": false,
        "cssrgb": "rgb(235,209,219)",
        "storeStripLocator": "102-C1",
        "similarColors": [
          "2255",
          "2262",
          "2269",
          "2662",
          "2535",
          "2276",
          "2658",
          "1974",
          "2543",
          "2531"
        ]
      },
      {
        "colorNumber": "6568",
        "coordinatingColors": {
          "coord2ColorId": 11244,
          "coord1ColorId": 2687,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2255",
        "isExterior": true,
        "isInterior": true,
        "name": "Lighthearted Pink",
        "rgb": 15586781,
        "lrv": 70.758,
        "brandedCollectionIds": [
          31797,
          31791
        ],
        "brandedCollectionNames": [
          "Pottery Barn Fall/Winter 2018",
          "Pottery Barn Fall/Winter 2018"
        ],
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 237,
        "green": 213,
        "blue": 221,
        "hue": 0.9444444444444445,
        "saturation": 0.3999999999999999,
        "lightness": 0.8823529411764706,
        "hex": "#edd5dd",
        "isDark": false,
        "cssrgb": "rgb(237,213,221)",
        "storeStripLocator": "103-C1",
        "similarColors": [
          "2248",
          "2262",
          "2269",
          "2276",
          "1974",
          "1981",
          "2658",
          "1995",
          "2662",
          "1988"
        ]
      }
    ]
  ]
}

export function getColors (): ColorSetPayload {
  return cloneDeep(Colors)
}

export function getColor (): Color | void {
  const colors = getColors()
  const fam1 = Object.keys(colors)[0]
  if (colors[fam1]) {
    return sample(colors[fam1][0])
  }
}
