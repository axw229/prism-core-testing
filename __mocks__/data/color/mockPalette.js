const mockPalette = {
  'colors': [
    {
      'colorNumber': '6765',
      'coordinatingColors': {
        'coord2ColorId': '11194',
        'coord1ColorId': '2870',
        'whiteColorId': '2162'
      },
      'description': [
        'Fairly Colorful',
        'Moderately Colorful',
        'Pale'
      ],
      'id': '2452',
      'isExterior': true,
      'isInterior': true,
      'name': 'Spa',
      'lrv': 64.476,
      'brandedCollectionNames': [],
      'colorFamilyNames': [
        'Blue'
      ],
      'brandKey': 'SW',
      'red': 167,
      'green': 220,
      'blue': 220,
      'hue': 0.5,
      'saturation': 0.43089430894308944,
      'lightness': 0.7588235294117647,
      'hex': '#a7dcdc',
      'isDark': false,
      'storeStripLocator': '164-C2',
      'similarColors': [
        '2634',
        '2630',
        '11225',
        '2472',
        '2445',
        '2622',
        '2459',
        '1606',
        '11226',
        '2451'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '6860',
      'coordinatingColors': {
        'coord2ColorId': '11339',
        'coord1ColorId': '11364',
        'whiteColorId': '2681'
      },
      'description': [
        'Soft',
        'Fairly Bright'
      ],
      'id': '2547',
      'isExterior': true,
      'isInterior': true,
      'name': 'Eros Pink',
      'lrv': 18.874,
      'brandedCollectionNames': [
        '2020 Play',
        '2019 Naturalist'
      ],
      'colorFamilyNames': [
        'Red'
      ],
      'brandKey': 'SW',
      'red': 200,
      'green': 79,
      'blue': 104,
      'hue': 0.9655647382920111,
      'saturation': 0.5238095238095238,
      'lightness': 0.5470588235294118,
      'hex': '#c84f68',
      'isDark': false,
      'storeStripLocator': '105-C5',
      'similarColors': [
        '2273',
        '2266',
        '2542',
        '2546',
        '1998',
        '2280',
        '2530',
        '2287',
        '2279',
        '30193'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '9018',
      'coordinatingColors': {
        'coord2ColorId': '11218',
        'coord1ColorId': '11200',
        'whiteColorId': '2366'
      },
      'description': [
        'Intense',
        'Pale'
      ],
      'id': '11195',
      'isExterior': false,
      'isInterior': true,
      'name': 'Honey Bees',
      'lrv': 70.14,
      'brandedCollectionNames': [
        '2018 Unity'
      ],
      'colorFamilyNames': [
        'Yellow'
      ],
      'brandKey': 'SW',
      'red': 251,
      'green': 214,
      'blue': 130,
      'hue': 0.11570247933884299,
      'saturation': 0.937984496124031,
      'lightness': 0.7470588235294118,
      'hex': '#fbd682',
      'isDark': false,
      'storeStripLocator': '134-C4',
      'similarColors': [
        '2382',
        '11199',
        '11196',
        '2588',
        '2375',
        '11198',
        '11194',
        '11191',
        '3146',
        '2361'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '9007',
      'coordinatingColors': {
        'coord2ColorId': '1666',
        'coord1ColorId': '2686',
        'whiteColorId': '2331'
      },
      'description': [
        'Rich',
        'Vibrant',
        'Pale'
      ],
      'id': '11184',
      'isExterior': false,
      'isInterior': true,
      'name': 'Melón Meloso',
      'lrv': 55.314,
      'brandedCollectionNames': [],
      'colorFamilyNames': [
        'Orange'
      ],
      'brandKey': 'SW',
      'red': 242,
      'green': 184,
      'blue': 140,
      'hue': 0.0718954248366013,
      'saturation': 0.7968749999999999,
      'lightness': 0.7490196078431373,
      'hex': '#f2b88c',
      'isDark': false,
      'storeStripLocator': '121-C3',
      'similarColors': [
        '2334',
        '2983',
        '2327',
        '11188',
        '2326',
        '2045',
        '2038',
        '3012',
        '2341',
        '2031'
      ],
      'ignore': false,
      'archived': false
    }
  ],
  'activeColor': {
    'colorNumber': '9007',
    'coordinatingColors': {
      'coord2ColorId': '1666',
      'coord1ColorId': '2686',
      'whiteColorId': '2331'
    },
    'description': [
      'Rich',
      'Vibrant',
      'Pale'
    ],
    'id': '11184',
    'isExterior': false,
    'isInterior': true,
    'name': 'Melón Meloso',
    'lrv': 55.314,
    'brandedCollectionNames': [],
    'colorFamilyNames': [
      'Orange'
    ],
    'brandKey': 'SW',
    'red': 242,
    'green': 184,
    'blue': 140,
    'hue': 0.0718954248366013,
    'saturation': 0.7968749999999999,
    'lightness': 0.7490196078431373,
    'hex': '#f2b88c',
    'isDark': false,
    'storeStripLocator': '121-C3',
    'similarColors': [
      '2334',
      '2983',
      '2327',
      '11188',
      '2326',
      '2045',
      '2038',
      '3012',
      '2341',
      '2031'
    ],
    'ignore': false,
    'archived': false
  },
  'toggleCompareColor': false,
  'temporaryActiveColor': null
}

export default mockPalette
