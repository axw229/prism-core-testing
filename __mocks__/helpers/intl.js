/**
 * Components using the react-intl module require access to the intl context.
 * This is not available when mounting single components in Enzyme.
 * These helper functions aim to address that and wrap a valid,
 * English-locale intl context around them.
 */

import React from 'react'
import { IntlProvider, intlShape } from 'react-intl'
import { shallow, mount } from 'enzyme'

const messages = require('src/translations/en-US.json') // en.json
const intl = new IntlProvider({ locale: 'en', messages }, {})
// const { intl } = intlProvider.getChildContext()

/**
 * When using React-Intl `injectIntl` on components, props.intl is required.
 */
const nodeWithIntlProp = node => {
  return React.cloneElement(<IntlProvider locale='en' messages={messages}>{node}</IntlProvider>, { intl })
}

export const shallowWithIntl = node => {
  return shallow(nodeWithIntlProp(node), { context: { intl } })
}

export default shallowWithIntl

export const mountWithIntl = node => {
  return mount(nodeWithIntlProp(node), {
    context: { intl },
    childContextTypes: { intl: intlShape }
  })
}
