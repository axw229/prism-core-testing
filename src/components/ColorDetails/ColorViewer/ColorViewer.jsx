// @flow
import React, { PureComponent } from 'react'

type Props = {
  color: Object
}

class ColorViewer extends PureComponent<Props> {
  render () {
    const { color } = this.props

    return (
      <React.Fragment>
        <strong>Color Viewer Color</strong>
        <hr />
        {JSON.stringify(color)}
        <hr />
      </React.Fragment>
    )
  }
}

export default ColorViewer
