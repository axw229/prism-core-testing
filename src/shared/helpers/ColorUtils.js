// @flow
import tinycolor from '@ctrl/tinycolor'
import memoizee from 'memoizee'
import { kebabCase, compact } from 'lodash'

import { ROUTE_PARAMS } from 'constants/globals'
import type { Color } from '../types/Colors'

/**
 * @param {*} colorValue String, hex, or tinycolor-compatible rgb(a)/hsl(a)/hsv(a) object
 * @param {number} newLuminosity number between 0 and 100
 * @description Takes a color in any tinycolor-compatible format and adjusts its luminosity to an absolute value between 0 and 100
 */
function _alterLuminosity (colorValue: any, newLuminosity: number): TinyColor {
  const tc = tinycolor(colorValue)
  const hsl = tc.toHsl()
  return tinycolor(`hsl(${hsl.h}, ${hsl.s * 100}%, ${newLuminosity}%)`)
}

export const alterLuminosity = memoizee(_alterLuminosity, function (colorValue, newLuminosity) {
  // cusom hash function so we can have unique memoized results based on both arguments
  return `${colorValue}_${newLuminosity}`
}, { primitive: true, length: 2 })

export const brightenAbsolute = alterLuminosity // alias
export const darkenAbsolute = alterLuminosity // alias

export const fullColorNumber = memoizee(function fullColorNumber (brandKey: string | void, colorNumber: string | void): string {
  let colorNamePrefix = colorNumber ? colorNumber + ' ' : ''

  if (colorNamePrefix && brandKey) {
    colorNamePrefix = brandKey + ' ' + colorNamePrefix
  }

  return colorNamePrefix
}, { primitive: true, length: 2 })

export const fullColorName = memoizee(function fullColorName (brandKey: string | void, colorNumber: string | void, name: string): string {
  return `${fullColorNumber(brandKey, colorNumber)}${name}`
}, { primitive: true, length: 3 })

// creates a CDP URL, this is used anywhere a CDP URL is needed so we're able to change the path of it in one location
export const generateColorDetailsPageUrl = memoizee(function generateColorDetailsPageUrl (color: Color): string {
  const colorNumber = kebabCase(fullColorNumber(color.brandKey, color.colorNumber))
  const colorName = kebabCase(color.name)

  return '/' + compact([
    ROUTE_PARAMS.ACTIVE,
    ROUTE_PARAMS.COLOR_DETAIL,
    color.id,
    `${colorNumber}-${colorName}`
  ]).join('/')
})

export const assembleColorWallPageUrl = memoizee(function assembleColorWallPageUrl (sectionName: string | void, familyName: string | void, colorId: string | void, colorSEO: string | void): string {
  const url = '/' + compact([
    ROUTE_PARAMS.ACTIVE,
    ROUTE_PARAMS.COLOR_WALL,
    sectionName ? `${ROUTE_PARAMS.SECTION}/${sectionName}` : null,
    familyName ? `${ROUTE_PARAMS.FAMILY}/${familyName}` : null,
    colorId ? `${ROUTE_PARAMS.COLOR}/${colorId}` : null,
    colorSEO ? `${colorSEO}` : null
  ]).join('/')

  if (url.slice(-1) === '/') {
    return url.slice(0, -1)
  }

  return url
}, { primitive: true, length: 3 })

export const generateColorWallPageUrl = memoizee(function generateColorWallPageUrl (sectionName: string | void, familyName: string | void, colorId: string | void, colorSEO: string | void): string {
  if (sectionName) {
    sectionName = kebabCase(sectionName)

    if (familyName) {
      familyName = kebabCase(familyName)
    } else {
      familyName = ''
    }

    if (colorId) {
      colorId = kebabCase(colorId)
      colorSEO = kebabCase(colorSEO)
    }
  } else {
    sectionName = familyName = colorId = colorSEO = ''
  }

  return assembleColorWallPageUrl(sectionName, familyName, colorId, colorSEO)
}, { primitive: true, length: 3 })
