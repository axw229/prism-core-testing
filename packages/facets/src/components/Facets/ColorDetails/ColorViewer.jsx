// @flow
import React, { useContext } from 'react'
import { FormattedMessage } from 'react-intl'
import { FEATURE_EXCLUSIONS } from '../../../constants/configurations'
import ConfigurationContext, { type ConfigurationContextType } from '../../../contexts/ConfigurationContext/ConfigurationContext'
import { fullColorNumber } from '../../../shared/helpers/ColorUtils'
import type { Color } from '../../../shared/types/Colors.js.flow'
import { shouldAllowFeature } from '../../../shared/utils/featureSwitch.util'

const baseClass = 'color-info'

type Props = { color: Color }

function ColorViewer ({ color }: Props) {
  const { colorWall: { colorSwatch = {} }, featureExclusions } = useContext<ConfigurationContextType>(ConfigurationContext)
  const { colorNumOnBottom = false, houseShaped = false } = colorSwatch
  const baseTitleClass = houseShaped ? `${baseClass}-house-shaped` : baseClass

  return (
    <div className={`${baseClass}__expanded-title`}>
      <h1 className={`${baseTitleClass}${colorNumOnBottom ? '__name-number' : '__number-name'}`}>
        <span className={`${baseTitleClass}__number`}>{fullColorNumber(color.brandKey, color.colorNumber)}</span>
        <span className={`${baseTitleClass}__name`}>{color.name}</span>
      </h1>
      {shouldAllowFeature(featureExclusions, FEATURE_EXCLUSIONS.colorDetailsSubtitles) && (
        <>
          <h2 className={`${baseClass}__type`}>
            {(color.isInterior) ? <FormattedMessage id='INTERIOR' /> : ''}
            {(color.isInterior && color.isExterior) ? ' / ' : ''}
            {(color.isExterior) ? <FormattedMessage id='EXTERIOR' /> : ''}
          </h2>
          {color.storeStripLocator && (
            <h3 className={`${baseClass}__rack-location`}>
              <FormattedMessage id='LOCATION_NUMBER' />: {color.storeStripLocator}
            </h3>
          )}
        </>
      )}
    </div>
  )
}

export default ColorViewer
