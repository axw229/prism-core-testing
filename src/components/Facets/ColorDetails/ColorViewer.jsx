// @flow
import React from 'react'
import { FormattedMessage } from 'react-intl'

import { fullColorNumber } from '../../../shared/helpers/ColorUtils'

type Props = {
  color: Object
}

function ColorViewer ({ color }: Props) {
  const BASE_CLASS = 'color-info'

  return (
    <div className={`${BASE_CLASS}__expanded-title`}>
      <h1 className={`${BASE_CLASS}__name-number`}>
        <span className={`${BASE_CLASS}__number`}>{fullColorNumber(color.brandKey, color.colorNumber)}</span>
        <span className={`${BASE_CLASS}__name`}>{color.name}</span>
      </h1>
      <h2 className={`${BASE_CLASS}__type`}>
        {(color.isInterior) ? <FormattedMessage id='INTERIOR' /> : ''}
        {(color.isInterior && color.isExterior) ? ' / ' : ''}
        {(color.isExterior) ? <FormattedMessage id='EXTERIOR' /> : ''}
      </h2>
      {color.storeStripLocator && (
        <h3 className={`${BASE_CLASS}__rack-location`}>
          <FormattedMessage id='LOCATION_NUMBER' />: {color.storeStripLocator}
        </h3>
      )}
    </div>
  )
}

export default ColorViewer
