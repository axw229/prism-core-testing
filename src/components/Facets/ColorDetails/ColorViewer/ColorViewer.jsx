// @flow
import React, { PureComponent } from 'react'
import { FormattedMessage } from 'react-intl'

import { fullColorNumber } from '../../../../shared/helpers/ColorUtils'

type Props = {
  color: Object
}

class ColorViewer extends PureComponent<Props> {
  static baseClass = 'color-info'

  render () {
    const { color } = this.props

    return (
      <React.Fragment>
        <div className={`${ColorViewer.baseClass}__expanded-title`}>
          <h1 className={`${ColorViewer.baseClass}__name-number`}>
            <span className={`${ColorViewer.baseClass}__number`}>{fullColorNumber(color.brandKey, color.colorNumber)}</span>
            <span className={`${ColorViewer.baseClass}__name`}>{color.name}</span>
          </h1>
          <h2 className={`${ColorViewer.baseClass}__type`}>
            {(color.isInterior) ? 'Interior' : ''}
            {(color.isInterior && color.isExterior) ? ' / ' : ''}
            {(color.isExterior) ? 'Exterior' : ''}
          </h2>
          <h3 className={`${ColorViewer.baseClass}__rack-location`}>
            <FormattedMessage id='LOCATION_NUMBER' />: {color.storeStripLocator}
          </h3>
        </div>
      </React.Fragment>
    )
  }
}

export default ColorViewer
