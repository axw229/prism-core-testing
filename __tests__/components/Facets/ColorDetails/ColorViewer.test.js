import React from 'react'
import { shallow } from 'enzyme'
import ColorViewer from 'src/components/Facets/ColorDetails/ColorViewer' 
import * as Colors from '__mocks__/data/color/Colors'
import { fullColorNumber } from 'src/shared/helpers/ColorUtils'

const color = Colors.getColor()
const BASE_CLASS = 'color-info'

const getColorViewer = (props) => {
    return shallow(<ColorViewer {...props} />)
}

describe('ColorViewer component without data', () => {
    let colorViewer
    beforeEach(() => {
        if(!colorViewer) {
            colorViewer = getColorViewer({color: []})
        }
    })

    it('ColorViewer Snapshot without data', () => {
        expect(colorViewer).toMatchSnapshot()
    })

    it('ColorViewer is rendering', () => {
        expect(colorViewer.exists).toBeTruthy()
    })

    it('ColorViewer is rendering empty span for number and name', () => {
        colorViewer.find('div > h1 > span').forEach((node) => {
            expect(node.text()).toEqual('')
        })
    })

    it('ColorViewer is rendering empty h2 for type', () => {
        expect(colorViewer.find('h2').text()).toEqual('')
    })
})

describe('ColorViewer component with data', () => {
    let colorViewer
    beforeEach(() => {
        if(!colorViewer) {
            colorViewer = getColorViewer({color: color})
        }
    })

    it('ColorViewer Snapshot with data', () => {
        expect(colorViewer).toMatchSnapshot()
    })

    it('ColorViewer is rendering', () => {
        expect(colorViewer.exists).toBeTruthy()
    })

    it('ColorViewer is rendering color number', () => {
        const colorNumberSpan = `div.${BASE_CLASS}__expanded-title h1.${BASE_CLASS}__name-number span.${BASE_CLASS}__number`
        const colorNumber = fullColorNumber(color.brandKey, color.colorNumber)
        expect(colorViewer.find(colorNumberSpan).contains(colorNumber)).toBeTruthy()
    })

    it('ColorViewer is rendering color name', () => {
        const colorNameSpan = `div.${BASE_CLASS}__expanded-title h1.${BASE_CLASS}__name-number span.${BASE_CLASS}__name`
        expect(colorViewer.find(colorNameSpan).contains(color.name)).toBeTruthy()
    })

    it('ColorViewer is rendering rack location', () => {
        const rackLocationHeader = `div.${BASE_CLASS}__expanded-title h3`
        expect(colorViewer.find(rackLocationHeader).contains(color.storeStripLocator))
    })
})