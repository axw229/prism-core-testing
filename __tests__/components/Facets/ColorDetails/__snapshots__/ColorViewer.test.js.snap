// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorViewer component with data ColorViewer Snapshot with data 1`] = `
<div
  className="color-info__expanded-title"
>
  <h1
    className="color-info__name-number"
  >
    <span
      className="color-info__number"
    >
      SW 6561 
    </span>
    <span
      className="color-info__name"
    >
      Teaberry
    </span>
  </h1>
  <h2
    className="color-info__type"
  >
    <FormattedMessage
      id="INTERIOR"
      values={Object {}}
    />
     / 
    <FormattedMessage
      id="EXTERIOR"
      values={Object {}}
    />
  </h2>
  <h3
    className="color-info__rack-location"
  >
    <FormattedMessage
      id="LOCATION_NUMBER"
      values={Object {}}
    />
    : 
    102-C1
  </h3>
</div>
`;

exports[`ColorViewer component without data ColorViewer Snapshot without data 1`] = `
<div
  className="color-info__expanded-title"
>
  <h1
    className="color-info__name-number"
  >
    <span
      className="color-info__number"
    />
    <span
      className="color-info__name"
    />
  </h1>
  <h2
    className="color-info__type"
  />
</div>
`;
